bridge/bridge.go

```diff
@@ -347,6 +458,111 @@ func (b *Bridge) newService(port ServicePort, isgroup bool) *Service {
+	// if internal, register the internal VIP services
+	if b.config.Internal {
+		for _, vip := range service.Endpoint.VirtualIPs {
+			if network, err := b.docker.NetworkInfo(vip.NetworkID); err != nil {
+				log.Println("unable to inspect network while evaluating VIPs for service:", service.Spec.Name, err)
+			} else {
+				// no point to publish docker swarm internal ingress network VIP
+				if network.Name != "ingress" && len(vip.Addr) > 0 && strings.Contains(vip.Addr, "/") {
+					vipAddr := strings.Split(vip.Addr, "/")[0]
+					if len(service.Endpoint.Ports) > 0 {
+						b.registerSwarmVipServicePorts(service.Spec.Name, true, vipAddr, service.Endpoint.Ports, service.Spec.TaskTemplate.ContainerSpec)
+					}
+				}
+			}
+		}
+	} else {
+		// if there is no published ports, no point to register it out side
+		if len(service.Endpoint.Ports) > 0 {
+			b.registerSwarmVipServicePorts(service.Spec.Name, false, b.config.HostIp, service.Endpoint.Ports, service.Spec.TaskTemplate.ContainerSpec)
+		}
+	}
+}
+
+// current implementation attempts to register VIP service every container add event
+// better way could be to listen for service create events, however according to
+// docker configuration there is no such events
+// registrations created here are unique, and not based on containers
+// so we will just create them and forget, i don't see proper way to cleanup them at the moment
+func (b *Bridge) registerSwarmVipServicePorts(serviceName string, inside bool, vip string, ports []swarm.PortConfig, config *swarm.ContainerSpec) {
+	for _, port := range ports {
+		b.registerSwarmVipService(serviceName, inside, vip, true, int(port.PublishedPort), port.Protocol, int(port.TargetPort), config)
+	}
+}
+
+func (b *Bridge) registerSwarmVipService(serviceName string, inside bool, vip string, isGroup bool, port int, protocol swarm.PortConfigProtocol, targetPort int, config *swarm.ContainerSpec) {
+
+	var tag string
+	if tag = "vip-outside"; inside {
+		tag = "vip-inside"
+	}
+
+	service := new(Service)
+	defaultName := serviceName + "-" + strconv.Itoa(port)
+
+	metadata, _ := swarmServiceMetaData(config, strconv.Itoa(targetPort))
+	service.Name = mapDefault(metadata, "name", defaultName)
+	// for _, env := range envs  {
+	// 	envSplited := strings.Split(env, "_")
+	// 	if len(envSplited) == 3 {
+	// 		if envSplited[0] == "SERVICE" {
+	// 			envPort, err := strconv.Atoi(envSplited[1])
+	// 			if err != nil {
+	// 				log.Println("Impossile to converse str to int", err)
+	// 			}
+	// 			if  envPort == targetPort {
+	// 				if strings.Split(envSplited[2], "=")[0] == "NAME" {
+	// 					service.Name = strings.Split(envSplited[2], "=")[1]
+	// 				} else if strings.Split(envSplited[2], "=")[0] == "TAGS" {
+	// 					tag = "vip-outside," + strings.Split(envSplited[2], "=")[1]
+	// 				}
+	// 			}
+	// 		}
+	// 	}
+	// }
+
+	if inside {
+		// VIP is global and singleton, so we can use service name as service id
+		service.ID = service.Name
```

Comment:
service.ID has to be in the format HOST_NAME:CONTAINER_NAME:EXPOSED_PORT. Otherwise it will not handled by `Cleanup()`. See https://github.com/alterway/registrator/blob/175e46ca15cc9b9a66ee1e61daa0f49e9b62b090/bridge/bridge.go#L160